packages/pwa/app/localization.config.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {defineMessage} from 'react-intl'
+import packageJson from '../package.json'
+
+export const localizationConfig = {
+    defaultLocale: packageJson.l10n.defaultLocale,
+    supportedLocales: [
+        {
+            id: 'en-GB',
+            message: defineMessage({defaultMessage: 'English (United Kingdom)'})
+        },
```

Comment:
We can extend this config object to support currencies too, for example.

Reply:
This file mimics other similar config files for Einstein and Commerce API.

Reply:
We had a meeting last week I think.. and there was a 3:1 or maybe 4:1 consensus that we want to aim to have a single point of configuration (aka 1 configuration file) for all the commerce api settings, proxies, localization and currency and whatever the else the future might bring.

Also, I wanted to open up conversation about this. But ideally we have a file that will allow use to validate it against a schema, those don't tent to be javascript files. But you're kinda of in a corner when it comes to this because of `defineMessages`. I'd like use to explore thinking about how we might be able to do this without javascript configuration files, or talk about why we need it to be javascript.

Reply:
+1 on the single config file with a predefined schema, that can be used for validations if needed in the future. What format are we thinking about - JSON, YAML, TOML? Is there any place where I can see what all config options would go in this file? Happy to pitch in if you need more hands.

Reply:
@sejal-salesforce here are all the files where we would extract the config options from:
- `pwa/package.json`: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/default-locale-selector-messages-W-9922981/packages/pwa/package.json
- and all of the `*.config.js` in this `pwa/app` folder: https://github.com/SalesforceCommerceCloud/pwa-kit/tree/default-locale-selector-messages-W-9922981/packages/pwa/app

Reply:
Just curious @vmarta. Could we move the default locale config into localization.config.js? Why leave this one property in package.json?

Reply:
As for which format to use for the 1 configuration file, my suggestion is to use javascript. Because it has some advantages over json format, and it can still be serialized into json. More details in this PR comment: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/143#discussion_r720526790

Reply:
@vcua-mobify that's because the default locale is also needed in package.json, in particular the npm scripts `compile-messages:pseudo` and `extract-messages`.

Reply:
@bendvc to move this PR forward, can we approach things in 2 steps?

- In this PR, we can have all localization-related configs in one file, which would make it easier for other developers to add/update locales in the meantime.
- And then furthermore in a separate PR, implement the one config file for the entire app.